File: 加密资料/SM加密算法.py
# # 实现sm 比较好的库gmssl
# from gmssl import sm2
#
# # 16 进制的公钥和私钥 private_key = '00B9AB0B828FF68872F21A837FC303668428DEA11DCD1B24429D0C99E24EED83D5' public_key =
# 'B9C9A6E04E9C91F7BA880429273747D7EF5DDEB0BB2FF6317EB00BEF331A83081A6994B8993F3F5D6EADDDB81872266C87C018FB4162F5AF347B483E24620207' sm2_crypt = sm2.CryptSM2(public_key=public_key, private_key=private_key)
#
# # 待加密数据和加密后数据为 bytes 类型
# data = b"this is the data to be encrypted"
# enc_data = sm2_crypt.encrypt(data)
# dec_data = sm2_crypt.decrypt(enc_data)
#
# print('enc_data: ', enc_data.hex())
# print('dec_data: ', dec_data)


# from gmssl import sm3, func
#
#
# def sm3_hash(message):
#     # python实现需要把编码数据转换成列表
#     hash_hex = sm3.sm3_hash(func.bytes_to_list(message))
#     print(hash_hex)
#
#
# # main
# if __name__ == '__main__':
#     message = b"123"  # bytes类型
#     sm3_hash(message)


# gmssl基于   bytes进行加密
from gmssl import sm4, func

# 创建SM4加密对象
sm4_crypt = sm4.CryptSM4()

key = b'0123456789ABCDEF0123456789ABCDEF'

# 设置密钥
sm4_crypt.set_key(key, sm4.SM4_ENCRYPT)

# 要加密的数据
data = b"Hello, SM4!"

# 加密数据
ciphertext = sm4_crypt.crypt_ecb(func.bytes_to_list(data))

# 将加密后的数据转换为字节串
encrypted_data = bytes(func.list_to_bytes(ciphertext))

# 解密数据（如果需要）
sm4_crypt.set_key(key, sm4.SM4_DECRYPT)
decrypted_data = sm4_crypt.crypt_ecb(ciphertext)
decrypted_data = bytes(func.list_to_bytes(decrypted_data))

print("原始数据:", data.decode("utf-8"))
print("加密后的数据:", encrypted_data.hex())
print("解密后的数据:", decrypted_data.decode("utf-8"))

File: 加密资料/sm加密算法.js
// sm-crypto   完整的库   node    成熟的库
// 非对称加密算法
// var sm2 = require('sm-crypto').sm2

// var keypair = sm2.generateKeyPairHex()
// var publickey = keypair.publicKey // 公钥
// var privatekey = keypair.privateKey // 私钥
//
// var text = 'python'
// enc_data = sm2.doEncrypt(text, publickey)
// console.log(enc_data)
// data = sm2.doDecrypt(enc_data, privatekey)
// console.log(data)

// 哈希
// var sm3 = require('sm-crypto').sm3
// var data = 'python'
// console.log(sm3(data).length);


// 对称加密算法
var sm4 = require('sm-crypto').sm4
var key = '0123456789ABCDEF0123456789ABCDEF'
var dta = 'python'

text = sm4.encrypt(dta, key)
dec_data = sm4.decrypt(text, key)

console.log(dec_data);
console.log(text)















